Modules/MSCloudLoginAssistant/MSCloudLoginAssistant.psm1

```diff
@@ -310,6 +315,85 @@ function Connect-M365Tenant
     }
 }
 
+function Compare-InputParametersForChange
+{
+    param (
+        [Parameter()]
+        [System.Collections.Hashtable]
+        $CurrentParamSet
+    )
+
+    $currentParameters = $currentParamSet
+    if ($null -ne $currentParameters['Credential'].UserName)
+    {
+        $currentParameters.Add('UserName', $currentParameters['Credential'].UserName)
+    }
+    $currentParameters.Remove('Workload') | Out-Null
+    $currentParameters.Remove('Credential') | Out-Null
+    $currentParameters.Remove('SkipModuleReload') | Out-Null
+    $currentParameters.Remove('UseModernAuth') | Out-Null
+    $currentParameters.Remove('ProfileName') | Out-Null
+    $currentParameters.Remove('Verbose') | Out-Null
+
+    $globalParameters = @{}
+
+
+    $workloadProfile = $Global:MSCloudLoginConnectionProfile
+
+    if ($null -eq $workloadProfile)
+    {
+        return $true
+    }
+    else
+    {
+        $workloadProfile = $Global:MSCloudLoginConnectionProfile.$Workload
```

Comment:
Where is the $Workload variable coming from? I don't see this being set anywhere.

Reply:
Good find. I modified the code a little bit.